### ISSUE TITLE
Option to collapse pre/post & asserts

### ISSUE THREAD

**Issue by olivergeorge** [cursive-ide/cursive#1406]
My asserts and pre/post conditions can seem quite bulky in small functions.  Perhaps editors should allow us to hide them.

```
(defn select-by [ms kvs]
  {:pre [(seq ms) (map? kvs)]}
  (-> (index ms (keys kvs)) (get kvs)))

(defn get-by [ms kvs]
  {:pre [(seq ms) (map? kvs)]}
  (let [matches (select-by ms kvs)]
    (if-not (next matches)
      (first matches)
      (assert (next matches) (str 'get-by " expects a unique match")))))
```

vs

```
(defn select-by [ms kvs]
  {:pre [...]}
  (-> (index ms (keys kvs)) (get kvs)))

(defn get-by [ms kvs] 
  {:pre [...]}
  (let [matches (select-by ms kvs)]
    (if-not (next matches)
      (first matches)
      (assert...))))
```

**Comment by olivergeorge** [comment]
Perhaps even colour would make all the difference.  Muted pre/post and assert perhaps?

**Comment by cursive-ide** [comment]
Good idea, I could probably automatically fold them, or at least have a config item to do that.

**Comment by olivergeorge** [comment]
Sounds great.

**Comment by olivergeorge** [comment]
Just wanted to bump this.  I'm finding that (s/assert ...) is becoming more common in my functions.  That makes things more verbose and I'd love a way to "fold them out" somehow.  

Yes, I want them and I don't want to see them...  how could I be more clear and reasonable!

How about this.  Allow s/assert statements to be presented as "mouse over hover to see spec".

So, as a random example picked from my code, this:

Would be displayed as this:
![image](https://user-images.githubusercontent.com/99447/35601295-5ca188bc-0686-11e8-9fde-5d4bc43f502b.png)

becomes this:

![image](https://user-images.githubusercontent.com/99447/35601315-7480fb66-0686-11e8-98a9-2645a27e130d.png)
![image](https://user-images.githubusercontent.com/99447/35601397-e65bcf90-0686-11e8-89c3-50797e22b505.png)
